src/sdk/trace/provider.zig

```diff
@@ -0,0 +1,26 @@
+const trace = @import("../../api/trace.zig");
+/// TracerProvider is an OpenTelemetry TracerProvider.
+/// That implements the TracerProvider interface.
+pub const TracerProvider = struct {
+    provider: trace.TracerProvider,
+
+    const Self = @This();
+
+    pub fn init() Self {
+        return Self{
+            .provider = trace.TracerProvider{
+                .tracer = tracer,
+            },
+        };
+    }
+
+    fn tracer(_: *trace.TracerProvider, _: []const u8, _: ?trace.TracerConfig) trace.Tracer {
+        // Get a pointer to the instance of the struct that implements the interface.
+        // const self: *Self = @fieldParentPtr("provider", iface);
+        return .{};
+    }
+};
```

Comment:
nice 👍🏼  `@fieldPartenPtr` is also how the metrics are implementing some of the interfaces, see https://github.com/zig-o11y/opentelemetry-sdk/blob/5e7dbb266b4045ece75df47eb36d37a5431704c8/src/metrics/exporter.zig#L21-L33

I like the methodology and as long as we document it explicitly, sgtm